rofl-proxy/src/http/tls.rs

```diff
@@ -207,6 +188,81 @@ impl CertificateProvisioner {
 
+    async fn init_acme(&self) -> instant_acme::Account {
+        match self.try_load_acme().await {
+            Ok(acme) => acme,
+            Err(err) => {
+                slog::info!(self.logger, "failed to load existing ACME account, provisioning new one";
+                    "err" => ?err,
+                );
+
+                // Back off to recover from Let's Encrypt authorization-failure limits.
+                // Limit: 10 new accounts per IP every 3 hours, refiling 1 slot every ~18 minutes.
+                // So we make the final interval >18 min.
+                // Rate-limited failures also count toward this limit.
+                //
+                // Note however, even with backoff, if multiple apps on the same host attempt
+                // to provision an ACME account concurrently, they may all hit this limit.
+                // Because rate-limited also count toward this limit this could create a never
+                // ending backoff situation.
+                let backoff = backoff::ExponentialBackoffBuilder::new()
+                    .with_initial_interval(Duration::from_secs(30))
+                    .with_multiplier(2.0)
+                    .with_randomization_factor(0.1)
+                    .with_max_interval(Duration::from_secs(20 * 60))
+                    .with_max_elapsed_time(Some(Duration::from_secs(60 * 60)))
+                    .build();
+                let result = backoff::future::retry(backoff, async || {
+                    let result = self.try_provision_acme().await;
+                    if let Err(ref err) = result {
+                        slog::error!(self.logger, "failed to provision ACME account";
+                            "err" => ?err,
+                        );
+                    }
+                    result.map_err(backoff::Error::transient)
+                })
+                .await;
+
+                // TODO: Give up after one hour? Or just keep retrying forever?
+                result.expect("failed to provision ACME account after retries")
+            }
+        }
+    }
+
+    /// Try to load previously created acme account.
+    async fn try_load_acme(&self) -> Result<instant_acme::Account> {
+        let data = fs::read_to_string(PERSISTENT_TLS_ACME_CREDENTIALS)?;
+        let creds: instant_acme::AccountCredentials = serde_json::from_str(&data)?;
+        let acc = instant_acme::Account::builder()?
+            .from_credentials(creds)
+            .await?;
+        Ok(acc)
+    }
+
+    /// Try to provision a new acme account.
+    async fn try_provision_acme(&self) -> Result<instant_acme::Account> {
+        let (acme, creds) = instant_acme::Account::builder()?
+            .create(
+                &instant_acme::NewAccount {
+                    contact: &[],
+                    terms_of_service_agreed: true,
+                    only_return_existing: false,
+                },
+                instant_acme::LetsEncrypt::Production.url().to_owned(),
```

Comment:
Maybe we should allow one to configure "staging" on Testnet or something? That has much lower rate limits.